services/blockstorage/block_sync_server.go

```diff
@@ -0,0 +1,86 @@
+package blockstorage
+
+import (
+	"errors"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
+	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+)
+
+func (s *service) sourceHandleBlockAvailabilityRequest(message *gossipmessages.BlockAvailabilityRequestMessage) error {
+	s.logger.Info("received block availability request", log.Stringable("sender", message.Sender))
+
+	lastCommittedBlockHeight := s.LastCommittedBlockHeight()
+
+	if lastCommittedBlockHeight <= message.SignedBatchRange.LastCommittedBlockHeight() {
+		return nil
+	}
+
+	firstAvailableBlockHeight := primitives.BlockHeight(1)
```

Comment:
will this always be true? if not, can this test somehow fail when the assumption changes?

Reply:
I will discuss this with Tal, its his logic and i agree with you that there might be problems here - in either way i will need to make a small change here to make it less error prone. this is relevant to the other two comments about the sync server